src/store/slice/auth.slice.js

```diff
@@ -18,12 +20,24 @@ export const authSlice = createSlice({
 		},
 		toggleLoading: (state, action) => {
 			return { ...state, loading: action.payload };
+		},
+		toggleDarkMode: (state, action) => {
+			if (action.payload)
+				document.documentElement.classList.add('dark');
+			else document.documentElement.classList.remove('dark');
+
+			toggleLocalTheme(action.payload);
+			return { ...state, darkMode: action.payload };
```

Comment:
we dont have to `return` state in reducers, you can just mutate the state safely. `redux-toolkit` will take care of it. Please remove all `return` statements from your reducers.